gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/testing/LocalGcdHelper.java

```diff
@@ -179,49 +179,134 @@ private static Path executablePath(String cmd) {
   }
 
   private static class ProcessStreamReader extends Thread {
-
-    private final Process process;
     private final BufferedReader reader;
 
-    ProcessStreamReader(Process process, String blockUntil) throws IOException {
+    ProcessStreamReader(InputStream inputStream) {
       super("Local GCD InputStream reader");
       setDaemon(true);
-      this.process = process;
-      reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+      reader = new BufferedReader(new InputStreamReader(inputStream));
+    }
+
+    void terminate() throws IOException {
+      reader.close();
```

Comment:
can you add to both stream readers a volatile flag `terminated` initialized to `false` and set to `true` in `terminate()` before `reader.close()` then make the code in the `run` method runs in `while (!terminated)`.